design/ceph/object-store-realm.md

```diff
@@ -0,0 +1,253 @@
+# Rook Object Store Realm
+
+## Overview
+
+Multisite replication between object stores can be configured to replicate the data in a realm over multiple Ceph clusters. 
+
+Creation of an object store realm allows for object stores from remote clusters to replicate to the cluster this resource is created and running on.
+
+### Prerequisites
+
+A Rook Ceph cluster with mons and osds must be up and running.
+
+### Ceph multi-site object store data model
+
+For reference, here is a description of the underlying Ceph Multisite data model.
+
+```
+A cluster has one or more realms
+
+A realm spans one or more clusters
+A realm has one or more zonegroups
+A realm has one master zonegroup
+A realm defined in another cluster is replicated with the pull command
+
+A zonegroup has one or more zones
+A zonegroup has one master zone
+A zonegroup spans one or more clusters
+A zonegroup defined in another cluster is replicated with the pull command
+A zonegroup defines a namespace for object IDs unique across its zones
+Zonegroup metadata is replicated to other zonegroups in the realm
+
+A zone belongs to one cluster
+A zone has a set of pools that store the user and object metadata and object data
+Zone data and metadata is replicated to other zones in the zonegroup
+A master zone needs to be created for secondary zones to pull from to replicate across zones
+```
+
+## Object Store Realm CRD
+
+The object store realm settings are exposed to Rook as a Custom Resource Definition (CRD). The CRD is the Kubernetes-native means by which the Rook operator can watch for new resources. The operator stays in a control loop to watch for a new object store realm, changes to an existing object store realm, or requests to delete an object store realm.
+
+### Bootstrap Realm
+
+To create a realm on a cluster the following section need to be configured.
+
+- `bootstrapRealm`: Initial configuration of the realm. This section contains a list of zonegroups to be created and configured inside the realm.
+- `zonegroups`: This section is inside of `boostrapRealm`. It is a list of zonegroups in the realm. Each zonegroup must have a name, an indication of whether it is the master zonegroup (isMaster), and the name of a `masterZone`. There can only be one master zonegroup in the realm.
+
+```yaml
+  bootstrapRealm:
```

Comment:
why do you need to call that bootstrap?
we can say that this operator only done initial setup (no updates), but we can still have this CRD as just "realm"

Reply:
+1